src/templates.ts

```diff
@@ -1,74 +1,135 @@
  *
  * @param resolvedConfig tailwind config
  * @param maxLevel maximum level of depth
  * @param nuxt nuxt app
+ * 
+ * @returns array of templates
  */
-export default function createTemplates (resolvedConfig: Partial<TWConfig> | ResolvedTwConfig, config: ExposeConfig, nuxt = useNuxt()) {
-  const dtsContent: Array<string> = []
+export default function createConfigTemplates(twConfig: Awaited<ReturnType<typeof loadTwConfig>>, config: ExposeConfig, nuxt = useNuxt()) {
+  const templates: ResolvedNuxtTemplate<any>[] = [];
+  const resolvedConfig = resolveConfig(twConfig.tailwindConfig as TWConfig)
+  const getTWConfig = () => import(twConfig.template.dst).then((config: TWConfig) => resolveConfig(config)).catch(() => resolvedConfig)
 
   const populateMap = (obj: any, path: string[] = [], level = 1) => {
     Object.entries(obj).forEach(([key, value = {} as any]) => {
-      const subpath = path.concat(key).join('/')
+      const subpathComponents = path.concat(key)
+      const subpath = subpathComponents.join('/')
 
       if (
         level >= config.level || // if recursive call is more than desired
         !isJSObject(value) || // if its not an object, no more recursion required
         Object.keys(value).find(k => !k.match(NON_ALPHANUMERIC_RE)) // object has non-alphanumeric property (unsafe var name)
       ) {
-        if (isJSObject(value)) {
-          const [validKeys, invalidKeys]: [string[], string[]] = [[], []]
-          Object.keys(value).forEach(i => (NON_ALPHANUMERIC_RE.test(i) ? validKeys : invalidKeys).push(i))
-
-          addTemplate({
-            filename: `tailwind.config/${subpath}.mjs`,
-            getContents: () => `${validKeys.map(i => `const _${i} = ${JSON.stringify(value[i])}`).join('\n')}\nconst config = { ${validKeys.map(i => `"${i}": _${i}, `).join('')}${invalidKeys.map(i => `"${i}": ${JSON.stringify(value[i])}, `).join('')} }\nexport { config as default${validKeys.length > 0 ? ', _' : ''}${validKeys.join(', _')} }`,
-            write: config.write
-          })
-          dtsContent.push(`declare module "${config.alias}/${subpath}" { ${validKeys.map(i => `export const _${i}: ${JSON.stringify(value[i])};`).join('')} const defaultExport: { ${validKeys.map(i => `"${i}": typeof _${i}, `).join('')}${invalidKeys.map(i => `"${i}": ${JSON.stringify(value[i])}, `).join('')} }; export default defaultExport; }`)
-        } else {
-          addTemplate({
-            filename: `tailwind.config/${subpath}.mjs`,
-            getContents: () => `export default ${JSON.stringify(value, null, 2)}`,
-            write: config.write
-          })
-          dtsContent.push(`declare module "${config.alias}/${subpath}" { const defaultExport: ${JSON.stringify(value)}; export default defaultExport; }`)
-        }
+        templates.push(addTemplate({
+          filename: `tailwind.config/${subpath}.mjs`,
+          options: { subpathComponents },
+          getContents: async ({ options }) => {
```

Comment:
again, all templates `getContents` have become `async`; this may be a bit more expensive, and this is also causing Nuxt to log compilation time for the templates.

Reply:
using `loadConfig` from tailwindcss now which gives slight performance benefit